templates/helm/values.yaml.tpl

```diff
@@ -28,6 +28,15 @@ deployment:
   # Which priorityClassName to set?
   # See: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority
   priorityClassName: ""
+
+  # Map of key/value pairs to be put in secret to act as a source for environment variables for the controller.
+  # Can be used to authenticate without IRSA (although IRSA is recommended)
+  # eg. AWS_ACCESS_KEY_ID: "YOUR_AWS_ACCESS_KEY_ID"
+  # eg. AWS_SECRET_ACCESS_KEY: "YOUR_AWS_SECRET_ACCESS_KEY"
+  secretEnvVars: {}
+
+  # Name of the secret to store your envvars, defaults to app fullname
+  secretName: ""
```

Comment:
OK, hear me out on this one... :)

How about we structure these configuration values like so:

```yaml
deployment:
  # env contains configuration for environment variables in the Deployment
  env:
    # fromSecret contains configuration for environment variables in the Deployment
    # read from a Secret that is automatically created by Helm
    fromSecret:
      # name is the name of the Secret that is created that contains environment
      # variables inserted into the Deployment (defaults to the application full name
      name: ""
      # vars contains a map of key/value pairs representing the environment variables
      # that will be inserted into the Secret and read into the Deployment
      vars: {}
```

with the above structure, we can neatly contain all the environment-variable setup into a sub-object. We may later want to add something like:

```yaml
deployment:
  # env contains configuration for environment variables in the Deployment
  env:
    # additionalVars is a map of key/value pairs to be added as-is to the Deployment
    additionalVars: {}
```

Reply:
that is way better, I'll update now :)

Reply:
This seems to just be replicating the `env` structure inside the Helm values schema. In that case, why not just inject values in the same way we do for annotations and labels? Essentially just take whatever structure they put under `deployment.env` and indent it. See [this example](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L11) for how to achieve it. Then we by default support anything that the deployment env can support.

Reply:
> This seems to just be replicating the `env` structure inside the Helm values schema. In that case, why not just inject values in the same way we do for annotations and labels? Essentially just take whatever structure they put under `deployment.env` and indent it. See [this example](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L11) for how to achieve it. Then we by default support anything that the deployment env can support.

Sorry, not following you @RedbackThomson. Are you suggesting just making `deployment.env` be an object with a schema matching the `Deployment.Spec.Env` struct from upstream?

Reply:
that would make it more flexible, happy with either, whichever you both prefer 😄

Reply:
> Sorry, not following you @RedbackThomson. Are you suggesting just making `deployment.env` be an object with a schema matching the `Deployment.Spec.Env` struct from upstream?

Sorry yeah it's hard to explain. Essentially the same way we treat [`serviceAccount.annotations`](https://github.com/aws-controllers-k8s/code-generator/blob/main/templates/helm/values.yaml.tpl#L76-L77) as a multi-line string, and then just [indent it](https://github.com/aws-controllers-k8s/code-generator/blob/main/templates/helm/templates/service-account.yaml#L15-L17) to put it into the manifest. I propose we do the same with the `deployment.env` field.  

This is the [common practice](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L143) for Helm value files to let people have ultimate customization over fields with arbitrary values.